core/domain/skill_domain.py

```diff
@@ -253,6 +253,11 @@ def require_valid_misconception_id(cls, misconception_id):
                 'Expected misconception ID to be an integer, received %s' %
                 misconception_id)
 
+        if misconception_id < 0:
+            raise utils.ValidationError(
+                'Expected misconception ID to be greater than 0, received %s' %
+                misconception_id)
```

Comment:
The error does not make sense, we don't want it to be > 0 but >= 0

Reply:
Also, should we add validation to the handler schema?

Reply:
Done. As for the handler schema, I'm not really sure if validation needs to be added there or not.

Reply:
I think it would be useful to add it there.

Reply:
As user cannot enter the misconception ID directly, I'm not able to find out where to add the validation in the handler schema. @vojtechjelinek @lkbhitesh07 Could you help me out here? Thanks

Reply:
Ah, okay if that is so then it is fine.